src/sentry/search/events/builder.py

```diff
@@ -1600,12 +1603,20 @@ def __init__(
         # Don't do any of the actions that would impact performance in anyway
         # Skips all indexer checks, and won't interact with clickhouse
         self.dry_run = dry_run
+        assert dataset is None or dataset in [Dataset.PerformanceMetrics, Dataset.Metrics]
         super().__init__(
-            # Dataset is always Metrics
-            Dataset.Metrics,
+            Dataset.Metrics if dataset is None else dataset,
```

Comment:
Why are you defaulting to the metrics dataset here ?

Reply:
Its an allowed option cause incident code is using the MetricQueryBuilder too,
And I defaulted to metrics to reduce the number of changes I need to make to incident code/tests, I'll comment this in the code too but longterm I plan to change this so the default is PerformanceMetrics instead.